### PR TITLE
SAK-49146 Common switch TypeImpl to nonstrict-read-write cache strategy and atomic

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/SAK-49146

Pending work. We were talking about this problem some months ago :)

### CODE REVIEW

**Review by ern — APPROVED** [review]
There will be db changes needed to drop the version in SAKAI_PERSON_T

Trying to remember if there were other changes needed?

**Comment by jesusmmp** [comment]
> There will be db changes needed to drop the version in SAKAI_PERSON_T
> 
> Trying to remember if there were other changes needed?

We talked about not drop the version in SAKAI_PERSON_T

Other changes were done here: https://github.com/sakaiproject/sakai/pull/11267/files

**Comment by ern** [comment]
if you remove the optimistic lock then version column is useless and should be removed.

**Review by ern — CHANGES_REQUESTED** [review]
Lets discuss this one in the next Core call

File: common/impl/src/java/org/sakaiproject/component/common/type/TypeImpl.hbm.xml [hunk 1]
```diff
@@ -3,9 +3,9 @@
 <hibernate-mapping package="org.sakaiproject.component.common.type">
 
   <class name="org.sakaiproject.component.common.type.TypeImpl"
-    table="CMN_TYPE_T" optimistic-lock="version">
+    table="CMN_TYPE_T">
```
Comment (ern) [inline common/impl/src/java/org/sakaiproject/component/common/type/TypeImpl.hbm.xml:6]:
```suggestion
    table="CMN_TYPE_T" optimistic-lock="version">
```

File: common/impl/src/java/org/sakaiproject/component/common/edu/person/SakaiPersonImpl.hbm.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <hibernate-mapping package="org.sakaiproject.component.common.edu.person">
 
   <class name="org.sakaiproject.component.common.edu.person.PersonImpl"
-    table="SAKAI_PERSON_T" optimistic-lock="version"
+    table="SAKAI_PERSON_T"
```
Comment (ern) [inline common/impl/src/java/org/sakaiproject/component/common/edu/person/SakaiPersonImpl.hbm.xml:6]:
```suggestion
    table="SAKAI_PERSON_T" optimistic-lock="version"
```

**Review by ern — APPROVED** [review]

**Pull request merged**